ImageInNavigationBarDemo/DemoTableViewController.swift

```diff
@@ -7,3 +7,112 @@
+
+        /// Height of NavBar for Small state. Usually it's just 44
+        static let NavBarHeightSmallState: CGFloat = 44
+        /// Height of NavBar for Large state. Usually it's just 96.5 but if you have a custom font for the title, please make sure to edit this value since it changes the height for Large state of NavBar
+        static let NavBarHeightLargeState: CGFloat = 96.5
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        navigationController?.navigationBar.prefersLargeTitles = true
+
+        title = "Large Title"
+
+        guard let navigationBar = self.navigationController?.navigationBar else { return }
+
+        // Initial setup for image for Large NavBar state since the the screen always has Large NavBar once it gets opened
+        navigationBar.addSubview(imageView)
+        imageView.layer.cornerRadius = 20.0
+        imageView.clipsToBounds = true
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            imageView.rightAnchor.constraint(equalTo: navigationBar.rightAnchor,
+                                             constant: -Const.ImageRightSpace),
+            imageView.bottomAnchor.constraint(equalTo: navigationBar.bottomAnchor,
+                                              constant: -Const.ImageBottomSpaceForLargeState),
+            imageView.heightAnchor.constraint(equalToConstant: Const.ImageSizeForLargeState),
+            imageView.widthAnchor.constraint(equalTo: imageView.heightAnchor)
+            ])
+    }
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 3
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 10
+    }
+
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "DemoCell", for: indexPath)
+        cell.textLabel?.text = "Sample"
+        return cell
+    }
+
+    // MARK: - Delegates
+
+    override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard let height = navigationController?.navigationBar.frame.height else { return }
+        moveAndResizeImage(for: height)
+    }
+
+    // MARK: - Private methods
+
+    private func moveAndResizeImage(for height: CGFloat) {
+        // Value of difference between icons for large and small states
+        let sizeDiff = Const.ImageSizeForLargeState * (1.0 - Const.SmallImageFactor) // 8.0
+
+        // This value = 14. It equals to difference of 12 and 6 (bottom spacing for large and small states)
+        // Also it adds 8.0 (size difference when the image gets smaller size)
+        let maxYTranslation = Const.ImageBottomSpaceForLargeState - Const.ImageBottomSpaceForSmallState + sizeDiff
+
+        if height <= Const.NavBarHeightSmallState {
```

Comment:
Looks great! I have one small improvement proposal here. I think that `if` branching here is redundant, all you need to do is bind your `scale`, `yTranslation` and `xTranslation` properties inside allowed range of values. Here is an example:
```swift
let yTranslation = max(0, min(maxYTranslation, (maxYTranslation - (coeff * Const.ImageBottomSpaceForSmallState + coeff * sizeDiff)))
```
Result is wrapped here in max and min calls to make boundaries.
This way you can rewrite your `xTranslation` and `scale` calculations and get rid of first 2 `if` branches.

Reply:
makes sense. thanks! my concern about this was that almost 100% of times user will go into 1st `if` statement that's why it didn't make much sense to calculate stuff every time. discussed this on the call and decided to test the code on iphone5 and see if it's not that consuming and won't create lags.